base/mac/conversions.h

```diff
@@ -0,0 +1,169 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BASE_MAC_CONVERSIONS_H_
+#define BRAVE_BASE_MAC_CONVERSIONS_H_
+
+#import <Foundation/Foundation.h>
+
+#include <string>
+#include <type_traits>
+#include <vector>
+
+/// This file is for converting from STL to Objective-C and vice-versa.
+/// Notes: Once we have C++17 support, we can remove all the template
+/// specializations and use `if constexpr` so that one function handles all
```

Comment:
this is allowed now https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++-features.md#constexpr-if-allowed

Reply:
Fixed to use `if-constexpr`